Modules/HTTP/Sources/HTTP/Middleware/ContentNegotiationMiddleware/ClientContentNegotiatonMiddleware.swift

```diff
@@ -0,0 +1,74 @@
+public struct ClientContentNegotiationMiddleware : Middleware {
+    public enum Mode {
+        case buffer
+        case stream
+    }
+
+    public let mode: Mode
+    public let types: [MediaTypeConvertible.Type]
+
+    var mediaTypes: [MediaType] {
+        return types.map({$0.mediaType})
```

Comment:
If I make it laze I have to make the functions that use mutating, and that's not nice. Instead I just made it a regular constant and assigned it at `init`.